beetsplug/absubmit.py

```diff
@@ -164,8 +164,13 @@ def _get_analysis(self, item):
                     item=item, error=e
                 )
                 return None
+            decoded = bytearray()
             with open(filename, 'r') as tmp_file:
```

Comment:
How about just opening the file in binary mode? `open(filename, 'rb')` should do it.